File: client-app/src/app/models/activity.ts
import { Profile } from "./profile";

export interface Activity {
    id: string;
    title: string;
    //date: string;
    date: Date | null;
    description: string;
    category: string;
    city: string;
    venue: string;

    // //'n
    // hostUserName?: string;
    // isCancelled?: boolean;
    // isGoing?: boolean;
    // isHost?: boolean;
    // host?: Profile;
    // activityAppUsers?: Profile[];

    //'n
    hostUserName: string;
    isCancelled: boolean;
    isGoing: boolean;
    isHost: boolean;
    host?: Profile;
    activityAppUsers: Profile[];
  }
  
  export class Activity implements Activity {
    constructor(activityFormValues?: ActivityFormValues) {
      Object.assign(this, activityFormValues);
    }
  }

  export class ActivityFormValues { //'n
    id?: string = undefined;
    title: string = '';
    date: Date | null = null;
    description: string = '';
    category: string = '';
    city: string = '';
    venue: string = '';

    constructor(activityFormValues?: ActivityFormValues) {
      if(activityFormValues) {
        this.id = activityFormValues.id;
        this.title = activityFormValues.title;
        this.date = activityFormValues.date;
        this.description = activityFormValues.description;
        this.category = activityFormValues.category;
        this.city = activityFormValues.city;
        this.venue = activityFormValues.venue;
      }
    }
  }

File: Domain/Activity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }

        //[Required]
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public bool IsCancelled { get; set; } //'n
        
        //public ICollection<AppUser> Attendees { get; set; }
        //public ICollection<ActivityAppUser> ActivityAppUsers { get; set; }
        public ICollection<ActivityAppUser> ActivityAppUsers { get; set; } = new List<ActivityAppUser>(); //'n
        //public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}